src/toxcore/dht_node.rs

```diff
@@ -95,6 +95,56 @@ impl DhtNode {
         self.kbucket.try_add(node)
     }
 
+
+    /**
+    Create a [`DhtPacket`](../dht/struct.DhtPacket.html) to peer with `peer_pk`
+    `PublicKey` containing a [`PingReq`](../dht/struct.PingReq.html) request.
+    */
+    fn create_ping_req(&self, peer_pk: &PublicKey) -> DhtPacket {
+        let ping = PingReq::new();
+        let shared_secret = &encrypt_precompute(peer_pk, &self.dht_secret_key);
```

Comment:
Add TODO: `precompute shared key to calc it 1 time`

Reply:
done